terraform/modules/asg/main.tf

```diff
@@ -0,0 +1,156 @@
+##
+# Data providers
+##
+data "aws_vpc" "selected" {
+  id = "${var.vpc_id}"
+}
+
+data "aws_subnet_ids" "app" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Layer       = "app"
+    stack       = "${var.stack}"
+    application = "${var.app}"
+  }
+}
+
+data "aws_subnet" "app" {
+  count = "${length(data.aws_subnet_ids.app.ids)}"
+  id    = "${data.aws_subnet_ids.app.ids[count.index]}"
+}
+
+data "aws_elb" "elb" {
+  name = "${var.elb_name}"
+}
+
+##
+# Security groups
+##
+resource "aws_security_group" "ci" {
```

Comment:
Response to https://github.com/CMSgov/bluebutton-web-deployment/pull/1238#discussion_r168809875:

> For the lower environments, particularly dev, I'd like CI to be able to deploy code refreshes (via Ansible playbook) on merge to master/develop, bypassing a full AMI build (it takes ~30 mins at the moment).

My only worry there would be if it's at all possible for a certain path of upgrades on a single instance would ever cause a bug that a new deployment wouldn't.  For something like a container this makes sense, so I guess I'm just saying this looks good assuming that break in isolation/reproducibility is not an issue.